Code/EnginePlugins/JoltPlugin/Constraints/Implementation/JoltGrabObjectComponent.cpp

```diff
@@ -99,25 +105,36 @@ bool ezJoltGrabObjectComponent::FindNearbyObject(ezGameObject*& out_pObject, ezT
   queryParam.m_uiCollisionLayer = m_uiCollisionLayer;
   queryParam.m_ShapeTypes = ezPhysicsShapeType::Static | ezPhysicsShapeType::Dynamic;
 
-  if (!pPhysicsModule->Raycast(hit, pOwner->GetGlobalPosition(), pOwner->GetGlobalDirForwards().GetNormalized(), m_fMaxGrabPointDistance * 5.0f, queryParam))
-    return false;
-
-  if (hit.m_fDistance > m_fMaxGrabPointDistance)
-    return false;
+  if (m_fCastRadius > 0.0f)
+  {
+    if (!pPhysicsModule->SweepTestSphere(hit, m_fCastRadius, pOwner->GetGlobalPosition(), pOwner->GetGlobalDirForwards().GetNormalized(), m_fMaxGrabPointDistance * 5.0f, queryParam))
+      return false;
+  }
+  else
+  {
+    if (!pPhysicsModule->Raycast(hit, pOwner->GetGlobalPosition(), pOwner->GetGlobalDirForwards().GetNormalized(), m_fMaxGrabPointDistance * 5.0f, queryParam))
+      return false;
+  }  
 
   const ezGameObject* pActorObj = nullptr;
   if (!GetWorld()->TryGetObject(hit.m_hActorObject, pActorObj))
     return false;
 
+  // If we hit a non-kinematic dynamic actor try to find a grab point.
```

Comment:
So is the idea that the grab component would be generally used for all interactions, including pressing buttons? So far we have done those separately with raycasts.

Reply:
Yep, we did the same in the PhysX implementation. I use that functionality in my game prototype.